.github/workflows/build.yml

```diff
@@ -53,3 +53,30 @@ jobs:
           }}/actions/runs/${{ github.run_id }}/cancel
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  test:
```

Comment:
In this setup, we actually run ubuntu integ tests twice - once in the `build` task and once here. There are 3 options here:

1. Reduce the matrix to only `windows-latest` and `macos-latest`. Dont really like this since it breaks symmetry. 
2. Remove the `integ` task from `build`. Dont really like this since it means running `npx projen build` locally will not run integ tests.
3. Patch the build job with a matrix. I think this is an elegant solution since it keeps everything in standard form and also runs the unit tests on all platforms (i guess this could be a problem if our unit tests are platform dependent, but i don't think they are, or should be). This is a feature I think should eventually go into projen, where the self-mutation step would run only after all the matrix jobs have passed.

Thoughts?

Reply:
Discussed offline - I've updated the integ tests to run in a separate workflow, which should resolve the self-mutation issues. I think it's okay that the ubuntu integ test still runs twice, since the one in the `build` workflow has snapshot updating enabled while the ones in `integ` don't. But I'm totally open to changing it.